src/app/3d/qgs3dmapcanvaswidget.cpp

```diff
@@ -209,14 +212,25 @@ Qgs3DMapCanvasDockWidget::Qgs3DMapCanvasDockWidget( QWidget *parent )
   layout->addWidget( mCanvas );
   layout->addWidget( mAnimationWidget );
 
-  contentsWidget->setLayout( layout );
-
-  setWidget( contentsWidget );
+  setLayout( layout );
 
   onTotalPendingJobsCountChanged();
+
+  mDockableWidget = new QgsDockableWidgetHelper( isDocked, mCanvasName, this, QgisApp::instance() );
+  toolBar->addWidget( mDockableWidget->toggleButton() );
+  connect( mDockableWidget, &QgsDockableWidgetHelper::closed, [ = ]()
+  {
+    QgisApp::instance()->close3DMapView( canvasName() );
+  } );
 }
 
-void Qgs3DMapCanvasDockWidget::saveAsImage()
+Qgs3DMapCanvasWidget::~Qgs3DMapCanvasWidget()
+{
+  mDockableWidget->setWidget( nullptr );
```

Comment:
wouldn't it be better to have this call in destructor of QgsDockableWidgetHelper, so that other classes don't have to call it too?

Reply:
True - Will do that 👍🏿